src/promptflow/promptflow/_sdk/_service/utils/utils.py

```diff
@@ -39,25 +40,40 @@ def get_port_from_config(create_if_not_exists=False):
     (HOME_PROMPT_FLOW_DIR / PF_SERVICE_PORT_FILE).touch(mode=read_write_by_user(), exist_ok=True)
     with open(HOME_PROMPT_FLOW_DIR / PF_SERVICE_PORT_FILE, "r", encoding=DEFAULT_ENCODING) as f:
         service_config = load_yaml(f) or {}
-        port = service_config.get("service", {}).get("port", None)
+        port = service_config.get(sys.executable, {}).get("port", None)
     if not port and create_if_not_exists:
         with open(HOME_PROMPT_FLOW_DIR / PF_SERVICE_PORT_FILE, "w", encoding=DEFAULT_ENCODING) as f:
             # Set random port to ~/.promptflow/pf.yaml
             port = get_random_port()
-            service_config["service"] = service_config.get("service", {})
-            service_config["service"]["port"] = port
+            service_config[sys.executable] = service_config.get(sys.executable, {})
+            service_config[sys.executable]["port"] = port
             dump_yaml(service_config, f)
     return port
 
 
+def kill_deprecated_service():
```

Comment:
why name this deprecated? 
just kill_service?

Reply:
it targets to kill original service started in this field 
![image](https://github.com/microsoft/promptflow/assets/26239730/e262c353-8b2a-4d98-90e9-42215e05e851)
, update to kill_service_get_from_service_field